src/Tribe/Tickets.php

```diff
@@ -2448,117 +2449,21 @@ public function send_tickets_email_for_attendees( $attendees, $args = [] ) {
 		 */
 		public function send_tickets_email_for_attendee( $to, $tickets, $args = [] ) {
-			// If no tickets to send for, do not send email.
-			if ( empty( $tickets ) ) {
-				return false;
-			}
-
-			if ( ! tec_tickets_emails_is_enabled() ) {
-				return $this->send_tickets_email_for_attendee_legacy( $to, $tickets, $args );
-			}
-
-			$defaults = [
-				'provider'      => 'ticket',
-				'post_id'       => 0,
-				'order_id'      => '',
-				'order_status'  => '',
-			];
-
-			// Set up the default arguments.
-			$args = wp_parse_args( $args, $defaults );
-
-			$provider      = $args['provider'];
-			$post_id       = $args['post_id'];
-			$order_id      = $args['order_id'];
-			$is_rsvp       = 'rsvp' === $provider || ( is_object( $provider ) && 'Tribe__Tickets__RSVP' === get_class( $provider ) );
-
-			if ( $is_rsvp ) {
-				if ( 'no' !== strtolower( $args['order_status'] ) ) {
-					$email_class      = tribe( TEC\Tickets\Emails\Email\RSVP::class );
-					$use_ticket_email = tribe_get_option( $email_class->get_option_key( 'use-ticket-email' ), false );
-					if ( ! empty( $use_ticket_email ) ) {
-						$email_class = tribe( TEC\Tickets\Emails\Email\Ticket::class );
-					}
-				} else {
-					$email_class = tribe( TEC\Tickets\Emails\Email\RSVP_Not_Going::class );
-				}
-
-			} else {
-				$email_class = tribe( TEC\Tickets\Emails\Email\Ticket::class );
-			}
-
-			if ( ! $email_class->is_enabled() ) {
-				return false;
-			}
-
-			// Filter the array so that we have a list of tickets by event.
-			$tickets_by_event = [];
-
-			foreach ( $tickets as $ticket ) {
-				$event_id = $ticket['event_id']; // @todo: check what happens with tickets from posts/pages.
-
-				if ( ! isset( $tickets_by_event[ $event_id ] ) ) {
-					$tickets_by_event[ $event_id ] = [];
-				}
-
-				$tickets_by_event[ $event_id ][] = $ticket;
-			}
-
-			// loop the tickets by event and send one email for each event.
-			foreach ( $tickets_by_event as $event_id => $event_tickets ) {
-				$email_class->set( 'post_id', $event_id );
-				$email_class->set( 'tickets', $event_tickets );
-				$email_class->recipient = $to;
-
-				$sent = $email_class->send();
-
-				// Handle marking the attendee ticket email as being sent.
-				if ( $sent ) {
-					// Mark attendee ticket email as being sent for each attendee ticket.
-					foreach ( $event_tickets as $attendee ) {
-						$this->update_ticket_sent_counter( $attendee['attendee_id'] );
+			/**
+			 * @since TBD
+			 *
+			 * @param null|mixed $pre     Determine if we should continue.
+			 * @param string     $to      The email to send the tickets to.
+			 * @param array      $tickets The list of tickets to send.
+			 * @param array      $args    The list of arguments to use for sending ticket emails.
+			 * @param static     $module  Instance of the Tickets Module.
+			 */
+			$pre = apply_filters( 'tec_tickets_send_tickets_email_for_attendee_pre', null, $to, $tickets, $args, $this );
```

Comment:
`pre` is a WP Core standards for when the filter is present just for short-circuiting a particular method:
Examples: `pre_update_option` or `pre_http_request`

Reply:
The nomenclature of WP usually has the `pre_` as a prefix of an action/procedure, not as a suffix. 

Still, this will become super confusing once we deprecate the old email delivery. Which is something that we plan to do soon. This method will have just a `pre_` filter with nothing else, so even the naming won't make sense.